src/Snapshot/main.vala

```diff
@@ -0,0 +1,124 @@
+#! /usr/bin/env -S vala workbench.vala --pkg libadwaita-1
+
+private Chessboard chessboard;
+private Gtk.Box box;
+private Gtk.Label label;
+
+public class Chessboard : Gtk.Widget {
+    const int SQUARE_SIZE = 70;
+    const int BOARD_SIZE = 8 * SQUARE_SIZE;
+    const float PIECE_SIZE = 1.5f * SQUARE_SIZE;
+
+    private Gdk.RGBA black_squares_color;
+    private Gdk.RGBA white_squares_color;
+    private Gdk.RGBA knight_color;
+
+    private float x;
+    private float y;
+    private float initial_x;
+    private float initial_y;
+    private Gsk.Path knight_path;
+    private Gsk.Stroke knight_stroke;
+    private Graphene.Rect knight_bounds;
+    private Gtk.GestureDrag gesture;
+    private bool gesture_started = false;
+
+    public Chessboard() {
```

Comment:
Looks great overall! We are just missing the space before the parentheses and this is ready to be merged.